src/reconcile/rec_visibility.c

```diff
@@ -223,9 +223,12 @@ static int
 __rec_find_and_save_delete_hs_upd(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins,
   WT_ROW *rip, WT_UPDATE_SELECT *upd_select)
 {
-    WT_UPDATE *delete_tombstone, *delete_upd;
+    WT_UPDATE *delete_tombstone, *delete_upd, *visible_all_upd;
```

Comment:
Is this part of the change still useful? If not, I'd prefer to leave it out.

Reply:
This code flow is useful. We are removing obsolete updates whenever we find them as part of the checkpoint. This is avoiding the page going through the normal update flow of obsolete check.